lib/magma/attribute.rb

```diff
@@ -26,6 +37,10 @@ def database_type
       nil
     end
 
+    def type
+      @type ||= self.class.name.match("Magma::(.*)Attribute")[1].underscore
```

Comment:
I added this to make `#update_option` a little easier to work with.

https://github.com/mountetna/magma/blob/e983166a3100529a81ddb1e3191509e54f7fc287/lib/magma/attribute.rb#L115-L134

It works but it feels a little brittle since it's derived from [`Magma::Model::ATTRIBUTE_TYPES`](https://github.com/mountetna/magma/blob/453cf160ffc7207a4ac7e25d2bb027ebc683e179/lib/magma/model.rb#L9-L12) and there's nothing really tying them together. It might also be a little confusing with `database_type`.

Reply:
I like `attribute_type` here. At some point I think we should build `ATTRIBUTE_TYPES` instead of hard-coding it; we could use this method to do so, which should tie them together nicely.

Reply:
When `type` moves into `EDITABLE_OPTIONS` what happens to line 127 here?

Reply:
> I like `attribute_type` here. At some point I think we should build `ATTRIBUTE_TYPES` instead of hard-coding it; we could use this method to do so, which should tie them together nicely.

Cool, I'll rename it to `attribute_type`. I was also playing around with dynamically building `ATTRIBUTE_TYPES`, but I backed off because I wasn't sure we wanted to go in that direction. I'll give it another try.

> When `type` moves into `EDITABLE_OPTIONS` what happens to line 127 here?

Hmm...I'm not following. When does `type` move into `EDITABLE_OPTIONS`?

Reply:
I made `attribute_type` a class method in cd71a7695caaa35b2d198d449296495258f994b0 so we can dynamically generate the `Magma::Model` attribute methods.

https://github.com/mountetna/magma/blob/cd71a7695caaa35b2d198d449296495258f994b0/lib/magma/model.rb#L10-L17